src/lib_json/json_writer.cpp

```diff
@@ -176,14 +177,11 @@ String valueToString(bool value) { return value ? "true" : "false"; }
 static bool isAnyCharRequiredQuoting(char const* s, size_t n) {
   assert(s || !n);
 
-  char const* const end = s + n;
-  for (char const* cur = s; cur < end; ++cur) {
-    if (*cur == '\\' || *cur == '\"' ||
-        static_cast<unsigned char>(*cur) < ' ' ||
-        static_cast<unsigned char>(*cur) >= 0x80)
-      return true;
-  }
-  return false;
+  return std::any_of(s, s + n, [](const char& cur) {
```

Comment:
`(unsigned char c)`

The name `cur` is longer and not helpfully so, since it's a 1-liner.
Save on some ugly static_casts. All comparisons we need can be done in the unsigned char domain.